packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts

```diff
@@ -123,13 +150,18 @@ export const transformPlotlyJsonToScatterChartProps = (
     const isXDate = isDateArray(xValues);
     const isXNumber = isNumberArray(xValues);
 
+    const { color: lineColor } = getColor(series.line?.color, index);
+    const { color: fillColor, opacity: fillOpacity } = getColor(series.fillcolor, index);
+
     return {
       legend: series.name || `Series ${index + 1}`,
       data: xValues.map((x: string | number, i: number) => ({
         x: isString ? (isXDate ? new Date(x) : isXNumber ? parseFloat(x as string) : x) : x,
         y: series.y[i],
       })),
-      color: series.line?.color || getNextColor(index),
+      color: lineColor,
+      fillColor,
+      opacity: fillOpacity,
```

Comment:
remove the opacity override in all places.

Reply:
then will we not consider 'a' in rgba?

Reply:
no. This could make the colors non accessible.